server/matchmaker/algorithm.py

```diff
@@ -42,39 +31,44 @@ def _unmatch(self, s1: Search):
 
 class StableMarriage(MatchmakingPolicy):
     def find(self) -> Dict[Search, Search]:
-        """Perform SM_NUM_TO_RANK runs of the stable matching algorithm. 
-        Assumes that _MatchingGraph.build_sparse() only returns edges whose matches are acceptable
-        to both parties."""
-        ranks = _MatchingGraph.build_sparse(self.searches)
+        """ Perform the stable matching algorithm until a maximal stable matching
+        is found. Assumes that _MatchingGraph.build_weighted() only returns edges
+        whose matches are acceptable to both parties.
+        """
+        ranks = _MatchingGraph.build_weighted(self.searches)
+        _MatchingGraph.remove_isolated(ranks)
+        max_degree = max((len(edges) for edges in ranks.values()), default=0)
+
         self.matches.clear()
 
-        for i in range(SM_NUM_TO_RANK):
+        for i in range(max_degree):
             self._logger.debug(
                 "Round %i of stable marriage, currently %i matches", i,
                 len(self.matches) // 2
             )
             # Do one round of proposals
-            if len(self.matches) == len(self.searches):
+            if len(self.matches) == len(ranks):
                 # Everyone found a match so we are done
```

Comment:
Depending on your style choices, maybe you can avoid the comment like that:
```python
everyone_found_a_match = (len(self.matches) == len(ranks))
if everyone_found_a_match:
  break
```
(the brackets are unnecessary but I feel they improve readability)

Reply:
More lines, more code. I disagree